src/Microsoft.AspNet.Authentication.OpenIdConnect/OpenIdConnectAuthenticationHandler.cs

```diff
@@ -217,6 +234,40 @@ public OpenIdConnectAuthenticationHandler(HttpClient backchannel)
 
             message.State = Options.StateDataFormat.Protect(properties);
 
+            if (Options.UsePostAuthorizationRequests)
+            {
+                using (var buffer = new MemoryStream())
+                using (var writer = new StreamWriter(buffer))
+                {
+                    var inputs = new StringBuilder();
+                    foreach (var parameter in message.Parameters)
+                    {
+                        var name = Options.HtmlEncoder.HtmlEncode(parameter.Key);
+                        var value = Options.HtmlEncoder.HtmlEncode(parameter.Value);
+
+                        inputs.AppendFormat(CultureInfo.InvariantCulture, InputTagFormat, name, value);
+                    }
+
+                    var issuer = Options.HtmlEncoder.HtmlEncode(message.IssuerAddress);
+
+                    writer.Write(string.Format(CultureInfo.InvariantCulture, HtmlFormat, issuer, inputs));
+                    writer.Flush();
+
+                    Response.ContentLength = buffer.Length;
+                    Response.ContentType = "text/html;charset=UTF-8";
+
+                    // Emit Cache-Control=no-cache to prevent client caching.
+                    Response.Headers.Set("Cache-Control", "no-cache");
+                    Response.Headers.Set("Pragma", "no-cache");
+                    Response.Headers.Set("Expires", "-1");
```

Comment:
Ah nice, I had not realized `Microsoft.Net.Http.Headers` was indirectly referenced. Thanks for the tip! :smile: